### PR TITLE
fix(mongo): join on column names with invalid characters

### PR DESCRIPTION
**Pull request by davinov**
Internal implementation of the mongo join translator requires a variable
name from each join column. Rules on variables names implies that
they cannot contain spaces an other non simple characters. Therefore, we
need to replace them by valid ones in the variable name.

### CODE REVIEW

**Comment by davinov** [comment]
Closes #576

File: src/lib/translators/mongo.ts [hunk 1]
```diff
@@ -53,6 +53,21 @@ function columnMap(colnames: string[]) {
   return _.fromPairs(colnames.map(col => [col, $$(col)]));
 }
 
+/**
+ * Generate a mongo [user variable](https://docs.mongodb.com/manual/reference/aggregation-variables/#user-variables)
+ * valid identifier from a column name.
+ *
+ * @param colname
+ */
+function columnToUserVariable(colname: string) {
```
Comment (PrettyWood) [inline src/lib/translators/mongo.ts:62]:
```suggestion
function columnToUserVariable(colname: string): string {
```
doesn't cost anything

File: tests/unit/mongo.spec.ts [hunk 2]
```diff
@@ -2485,6 +2485,74 @@ describe('Pipeline to mongo translator', () => {
...
+        right_pipeline: rightPipeline,
+        type: 'inner',
+        on: [
+          ['some column', 'some column'],
+          ['other column', 'other column'],
+        ],
+      },
+    ];
+    const querySteps = mongo36translator.translate(pipeline);
+    expect(querySteps).toEqual([
```
Comment (PrettyWood) [inline tests/unit/mongo.spec.ts:2514]:
```suggestion
    expect(querySteps).toStrictEqual([
```
doesn't cost anything as well

File: src/lib/translators/mongo.ts [hunk 3]
```diff
@@ -53,6 +53,21 @@ function columnMap(colnames: string[]) {
   return _.fromPairs(colnames.map(col => [col, $$(col)]));
 }
 
+/**
+ * Generate a mongo [user variable](https://docs.mongodb.com/manual/reference/aggregation-variables/#user-variables)
```
Comment (PrettyWood) [inline src/lib/translators/mongo.ts:57]: 👍 top !

**Pull request merged**